dotCMS/src/main/java/com/dotcms/rest/api/v1/content/ContentResource.java

```diff
@@ -232,4 +224,77 @@ private Contentlet getContentlet(final String inode,
         DotPreconditions.notNull(contentlet, ()-> "contentlet-was-not-found", DoesNotExistException.class);
         return contentlet;
     }
+
+    /**
+     * Will return a Contentlet objects. If you are building large pulls and depending on the types
+     * of fields on the content this can get expensive especially with large data sets.<br />
+     * EXAMPLE:<br /> /api/v1/content/related<br />
+     *
+     * {
+     *     "identifier":"d66309a7378bbad381fda3accd7b2e80",
+     *     "fieldVariable":"myRelationship",
+     *     "condition":"+title:child5",
+     *     "orderBy":"title desc"
+     * }
+     * The method will figure out language, working and
+     * live for you if not passed in with the condition Returns empty List if no results are found
+     * @param request  - Http Request
+     * @param response - Http Response
+     * @param pullRelatedForm {@link PullRelatedForm}
+     * @return Returns empty List if no results are found
+     */
+    @POST
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    @Path("related")
+    public Response pullRelated(@Context final HttpServletRequest request,
```

Comment:
Note to us - we need to begin returning structured objects and not just returning `javax.ws.rs.core.Response` so OpenApi can understand the return that is expected.  This is ok as is but future forward I think we need to do a bit more work here.

Also, how does this look in swagger?  Does it need some annotations?

Reply:
@jdotcms take a look at this for needed Swagger annotations https://github.com/dotCMS/core/blob/9e7d22db567b68ea0f54ee8526bbb1a51e45a556/dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java#L1489